lib/board.rb

```diff
@@ -28,46 +28,87 @@ def initialize
+    coordinates.all? { |coord| valid_coordinate?(coord) }
+  end
+
+#ADD helper method to parse coordinates into rows
+  def parse_rows(coordinates)
+    rows = []
+    coordinates.each do |coordinate|
+      rows << coordinate[0]
+    end
+    return rows
+  end
+
+#ADD helper method to parse coordinates into columns
+  def parse_columns(coordinates)
+    columns = []
+    coordinates.each do |coordinate|
+      columns << coordinate[1].to_i
+    end
+    return columns
+  end
+
   def valid_placement?(ship, coordinates)
     # Method checks whether number of elements is equal to ship length
     # and then verifies that all coordinates are valid
-    if ship.length == coordinates.count && !overlap?(coordinates) # overlap? is negated
-      if coordinates.all? { |coord| valid_coordinate?(coord) }
+
+
+    # #START CODE COMMENT
+    # if ship.length == coordinates.count && !overlap?(coordinates) # overlap? is negated
+    #   if valid_coordinates?(coordinates)
+
       # This section parses the input coordinates and creates
       # two separate arrays of rows and columns.
       # NOTE: could be refactored into separate helper method later
-        rows = []
-        columns = []
-        coordinates.each do |coordinate|
-          rows << coordinate[0]
-          columns << coordinate[1].to_i
-        end
+
+      #START CODE COMMENT
+        # rows = []
+        # columns = []
+        # coordinates.each do |coordinate|
+        #   rows << coordinate[0]
+        #   columns << coordinate[1].to_i
+        # end
 
       # Conditional flow control begins here:
       # Logic is either rows or columns must be consecutive
       # and the other must have static values. For example:
       # In the coordinates [A1, A2, A3], row (A) stays static and column (1, 2, 3) increments
       # Note: Coordinates will be entered L->R or T->B order
-        if check_consecutive(ship, rows) && check_static(columns)
-          true
-        elsif check_consecutive(ship, columns) && check_static(rows)
-          true
-        else
-          false
-        end
-      else
-        false
-      end
-    else
-      false
-    end
+
+      #START CODE COMMENT
+    #     if check_consecutive(ship, rows) && check_static(columns)
+    #       true
+    #     elsif check_consecutive(ship, columns) && check_static(rows)
+    #       true
+    #     else
+    #       false
+    #     end
+    #   else
+    #     false
+    #   end
+    # else
+    #   false
+    # end
+
+    #REPLACEMENT USING HELPER METHODS:
+    length_correct = (ship.length == coordinates.count)
+    no_overlap = !overlap?(coordinates)
+    on_board = valid_coordinates?(coordinates)
+    parsed_rows = parse_rows(coordinates)
+    parsed_columns = parse_columns(coordinates)
+    horizontal_consecutive = check_consecutive(ship, parsed_columns) && check_static(parsed_rows)
+    vertical_consecutive = check_consecutive(ship, parsed_rows) && check_static(parsed_columns)
```

Comment:
Wondering if all these variables that return booleans should have `?`s on the end? 

Similarly wondering if we should rename these: 
- check_static --> static?
- check_consecutive --> consecutive?

Reply:
Yeah, I like that change. I can implement in the next PR